apps/ledger-live-desktop/src/renderer/modals/CreateLocalManifest/index.tsx

```diff
@@ -0,0 +1,371 @@
+} from "@ledgerhq/live-common/platform/types";
+import Text from "~/renderer/components/Text";
+import { useLocalLiveAppContext } from "@ledgerhq/live-common/platform/providers/LocalLiveAppProvider/index";
+import Switch from "~/renderer/components/Switch";
+import FormLiveAppInput from "./FormLiveAppInput";
+import NestedFormCategory from "./NestedFormCategory";
+import updateObjectAtPath from "lodash/set";
+import cloneDeep from "lodash/cloneDeep";
+import FormLiveAppSelector from "./FormLiveAppSelector";
+import FormLiveAppArrayInput from "./FormLiveAppArrayInput";
+import { useCategories, useWalletAPICurrencies } from "@ledgerhq/live-common/wallet-api/react";
+import { WalletAPICurrency } from "@ledgerhq/live-common/wallet-api/types";
+
+import { useManifests } from "@ledgerhq/live-common/platform/providers/RemoteLiveAppProvider/index";
+import FormLiveAppArraySelect from "./FormLiveAppArraySelect";
+import { Separator } from "~/renderer/components/Onboarding/Screens/SelectUseCase/Separator";
+import { DEFAULT_FORM, DEFAULT_VALUES } from "./defaultValues";
+
+function createLocalManifest() {
+  return (
+    <Modal
+      width={600}
+      name="MODAL_CREATE_LOCAL_APP"
+      centered
+      render={({ data, onClose }) => <FormLocalManifest data={data} onClose={onClose} />}
+    />
+  );
+}
+
+function FormLocalManifest({
+  onClose,
+}: {
+  data: { manifest?: LiveAppManifest };
+  onClose: () => void;
+}) {
+  const { addLocalManifest } = useLocalLiveAppContext();
+  const [form, setForm] = useState<LiveAppManifest>(cloneDeep(DEFAULT_FORM));
+  const [isDapp, setIsDapp] = useState("dapp" in form);
+
+  const completeManifests = useManifests({ visibility: ["complete"] });
+  const { categories } = useCategories(completeManifests);
+  const walletAPICurrencies: WalletAPICurrency[] = useWalletAPICurrencies();
+
+  DEFAULT_VALUES.categories = categories.filter((category: string) => category !== "all");
+  DEFAULT_VALUES.currencies = walletAPICurrencies.map(currency => currency.id);
+
+  const handleSwitchEthDapp = () => {
+    setIsDapp(prevState => !prevState);
+    setForm((prevState: LiveAppManifest) => {
+      !isDapp
+        ? (prevState.dapp = cloneDeep(DEFAULT_FORM.dapp) as LiveAppManifestDapp)
+        : delete prevState.dapp;
+      return prevState;
+    });
+  };
+
+  const formIsValid = useMemo(() => LiveAppManifestSchema.safeParse(form).success, [form]);
+
+  const handleChange = useCallback((path: string, value: unknown) => {
+    setForm((prevState: LiveAppManifest) => {
+      return updateObjectAtPath({ ...prevState }, path, value);
+    });
+  }, []);
+
+  const submitHandler = (e: KeyboardEvent) => {
+    e.preventDefault();
+    if (e.detail !== 0) {
+      formIsValid && addLocalManifest(form);
+      onClose();
+    }
+  };
+
+  const contentInput = useCallback(() => {
+    return Object.keys(form.content).map(uncastedContentKey => {
+      const contentKey = uncastedContentKey as keyof LiveAppManifest["content"];
```

Comment:
We can create an util like the one for [`.includes`](https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledger-live-common/src/helpers.ts) in the same way as this lib: https://github.com/sindresorhus/ts-extras/blob/main/source/object-keys.ts#L24 
So you can avoid the manual `as keyof`